CONTRIBUTING.md

```diff
@@ -135,6 +135,30 @@ Most features will likely have most of the code in the `containerd-shim-wasm` pr
 
 A tip for developing a new feature is to implement it and test it with one runtime you are familiar with then add it to all the runtimes.  This makes it easier to test and iterate before making changes across all the runtimes.
 
+## Adding new shims
+
+We welcome new shims, though you can also host them in your own repositories as well and use the `coantainerd-shim-wasm` crate.  We recognize that the project is moving fast and having them in this repository can reduce the need for changes as well for discoverability.  
+
+Please open an issue before submitting a PR for discussion to make sure it is a good fit.  As a general rule, we want shims to be adopting WASI spec (this is after all called run*wasi* :-)). In the future we may require shims in the repository to pass WASI compliance tests when the standards mature more. See https://github.com/containerd/runwasi/issues/338 for more discussion.
+
+Since we are not experts in every runtime we also need a commitment from the runtime owners to contribute to maintenance of the shim.
+
+## Removing Shims
+
+This is a fast moving space, with lots of innovation happening and some shims may eventually need to be removed.
```

Comment:
One question that I brough up in the community call today is how to deal with scenarios where immediate actions are required. For example, say we introduced a breaking change in the core crate and we only implement that in one of the shims, and then obviously the CI/CD pipelines are going to break for other runtimes. How to resolve this scenario so that we can quickly iterate through new features and experiments?

Reply:
Great point. I will add something about this. My take is it would be case by case basis but if we can't resolve something and it is blocking then disabling and opening an issue to track (and potentially kick off the removal if no responses) fixing it.

Reply:
added a section for this